R/hardcode.R

```diff
@@ -0,0 +1,131 @@
+#' @importFrom rlang :=
+#' @keywords internal
+sdtm_hardcode <- function(raw_dat,
+                          raw_var,
+                          tgt_var,
+                          tgt_val,
+                          tgt_dat = raw_dat,
+                          by = NULL,
+                          ct = NULL,
+                          cl = NULL) {
+
+  # TODO: Assertions.
+
+  tgt_val <- ct_map(tgt_val, ct = ct, cl = cl)
+
+  raw_dat |>
+    dplyr::right_join(y = tgt_dat, by = by) |>
+    dplyr::mutate("{tgt_var}" := recode(x = !!rlang::sym(raw_var), to = tgt_val)) |>
+    dplyr::select(-rlang::sym(raw_var)) |>
+    dplyr::relocate(tgt_var, .after = dplyr::last_col())
```

Comment:
Isn't the derived variable already the last one?

Reply:
Not always, namely when the derived variable is already existing and might be placed in some other order other than the last.

Reply:
Not applicable anymore.